### DISCUSSION TITLE
Customize an existing swiperjs slider

### DISCUSSION THREAD

**Discussion by datgausaigon** [elementor/elementor#23769] [Developers Community]
Hi, 
As we know, Elementor uses swiperjs for Sliders and Carousels.

Starting in Elementor 2.9.0, the Swiper instance will now be available in the cache of this element’s jQuery instance.
Read the following link: [https://developers.elementor.com/expose-swiper-in-elementor/](https://developers.elementor.com/expose-swiper-in-elementor/)

I want to be able to customize Slider and Carousel in the above way. I tried changing the "effect" of a "Carousel Widget" with the following code (This code is created with Elementor/Custom Code, attaches End of Body, must have jQuery, located in <script></script>) :

```
(function ($) {

    'use strict';

    $(document).ready(function () {
        $(window).on('elementor/frontend/init', function() {
            
            elementorFrontend.hooks.addAction('frontend/element_ready/global', function($scope, $) {
                
                if($scope.attr('id') === 'my-id') {
    
                    var imageCarousel = $scope.find('.swiper-container');
                    var swiperInstance = imageCarousel.data('swiper');
                    
                    if(swiperInstance) {
                        swiperInstance.params.effect = 'fade';
                        swiperInstance.update();
                    }
                }
            });
        });
    });

})(jQuery);
```

"Carousel Widget" is added "ID: my-id" using "Advanced/CSS ID" settings.

But it doesn't work, anyone have any way to make it work? Where is it wrong?

Thanks ❤️

**Comment by perks98** [comment]
I am also interested in this topic. If you get it to work please let me know. For me when I try and edit a swiper in the developer console it works but when running in a script I get a 'params' error.
I'm not the best with JavaScript, but when I added 'console.log' in a few areas of your code it seems the actions in 'if(swiperInstance)' don't fire. That's all I know.

**Reply by datgausaigon** [reply]
Yep, you are right!
It's cannot goto "elementorFrontend.hooks.addAction('frontend/element_ready/global'" and access swiperjs instance.

**Comment by perks98** [comment]
@datverse so for me the code runs before Swiper gets defined, even if the code is placed at the most bottom of the body tag. So in your code's use case I've made it wait for Swiper to be defined so it waits every 0.1 second but it is async so it doesn't interrupt the code, not sure if this code helps you?

```
window.addEventListener( 'elementor/frontend/init', () => {
    jQuery(document).ready(function( $ ) {
        (async() => {
            console.log("waiting for swiper");
            while(typeof Swiper === "undefined") 
                await new Promise(resolve => setTimeout(resolve, 100));
            console.log("swiper is now defined");
            const imageCarousel = $( '#my-id .swiper' ),
            swiperInstance = imageCarousel.data( 'swiper' );
            swiperInstance.params.effect = 'fade';
            swiperInstance.update();
            console.log("swiper custom code finished");
        })();
        console.log("above code doesn't block main function stack while waiting for swiper");
    })
} ); 
```
But for me the fade effect doesn't do anything.
